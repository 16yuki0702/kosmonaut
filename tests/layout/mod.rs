#[cfg(test)]
mod tests {
    use crate::util::{dump_layout_cmd, dump_layout_cmd_scaled};
    use insta::assert_snapshot;

    #[test]
    fn rainbow_divs_baseline() {
        let mut dump_layout_cmd = dump_layout_cmd();
        dump_layout_cmd
            .arg("--files")
            .arg("tests/websrc/rainbow-divs.html")
            .arg("tests/websrc/rainbow-divs.css")
            .succeeds()
            .no_stderr();
        assert_snapshot!(dump_layout_cmd.stdout());
    }

    #[test]
    fn rainbow_divs_baseline_two_scale_factor() {
        let mut dump_layout_cmd = dump_layout_cmd_scaled(2.0);
        dump_layout_cmd
            .arg("--files")
            .arg("tests/websrc/rainbow-divs.html")
            .arg("tests/websrc/rainbow-divs.css")
            .succeeds()
            .no_stderr();
        assert_snapshot!(dump_layout_cmd.stdout());
    }
}
