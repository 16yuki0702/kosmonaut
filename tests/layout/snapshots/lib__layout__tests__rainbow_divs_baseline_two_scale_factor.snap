---
source: tests/layout/mod.rs
expression: dump_layout_cmd.stdout()
---
DOCUMENT Inline LayoutBox at (0, 0) size 1920x368
  DOCTYPE Inline LayoutBox at (0, 0) size 1920x0
  HTML Block LayoutBox at (0, 0) size 1920x368
     Anonymous LayoutBox at (0, 0) size 1920x0
      TEXT Inline LayoutBox at (0, 0) size 1920x0
    BODY Block LayoutBox at (16, 16) size 1888x336
       Anonymous LayoutBox at (0, 0) size 1856x0
        TEXT Inline LayoutBox at (0, 0) size 1856x0
      DIV Block LayoutBox at (40, 40) size 1840x288
         Anonymous LayoutBox at (0, 0) size 1792x0
          TEXT Inline LayoutBox at (0, 0) size 1792x0
        DIV Block LayoutBox at (64, 64) size 1792x240
           Anonymous LayoutBox at (0, 0) size 1744x0
            TEXT Inline LayoutBox at (0, 0) size 1744x0
          DIV Block LayoutBox at (88, 88) size 1744x192
             Anonymous LayoutBox at (0, 0) size 1696x0
              TEXT Inline LayoutBox at (0, 0) size 1696x0
            DIV Block LayoutBox at (112, 112) size 1696x144
               Anonymous LayoutBox at (0, 0) size 1648x0
                TEXT Inline LayoutBox at (0, 0) size 1648x0
              DIV Block LayoutBox at (136, 136) size 1648x96
                 Anonymous LayoutBox at (0, 0) size 1600x0
                  TEXT Inline LayoutBox at (0, 0) size 1600x0
                DIV Block LayoutBox at (160, 160) size 1600x48
                   Anonymous LayoutBox at (0, 0) size 1552x0
                    TEXT Inline LayoutBox at (0, 0) size 1552x0
                  DIV Block LayoutBox at (184, 184) size 1552x0
                     Anonymous LayoutBox at (0, 0) size 1504x0
                      TEXT Inline LayoutBox at (0, 0) size 1504x0
                   Anonymous LayoutBox at (0, 0) size 1552x0
                    TEXT Inline LayoutBox at (0, 0) size 1552x0
                 Anonymous LayoutBox at (0, 0) size 1600x0
                  TEXT Inline LayoutBox at (0, 0) size 1600x0
               Anonymous LayoutBox at (0, 0) size 1648x0
                TEXT Inline LayoutBox at (0, 0) size 1648x0
             Anonymous LayoutBox at (0, 0) size 1696x0
              TEXT Inline LayoutBox at (0, 0) size 1696x0
           Anonymous LayoutBox at (0, 0) size 1744x0
            TEXT Inline LayoutBox at (0, 0) size 1744x0
         Anonymous LayoutBox at (0, 0) size 1792x0
          TEXT Inline LayoutBox at (0, 0) size 1792x0
       Anonymous LayoutBox at (0, 0) size 1856x0
        TEXT Inline LayoutBox at (0, 0) size 1856x0

